gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/AbstractUserService.java

```diff
@@ -176,6 +181,8 @@ public Completable delete(ReferenceType referenceType, String referenceId, Strin
                         .andThen((ReferenceType.ORGANIZATION != referenceType) ? Completable.complete() :
                                 membershipService.findByMember(userId, MemberType.USER)
                                         .flatMapCompletable(membership -> membershipService.delete(membership.getId())))
+                        .andThen(passwordHistoryService.findUserHistory(referenceType, referenceId, userId)
```

Comment:
A single `deleteByUser` method should be more efficient.

Reply:
Where would that live? In `AbstractUserService?`?

Reply:
To me it is a method into the PasswordHistoryService

Reply:
Ah I thought you were talking about including the other delete calls here. Makes sense.